internal/pkg/errorconcept/device_profile.go

```diff
@@ -114,6 +115,20 @@ func (r deviceProfileInvalidState_StatusConflict) message(err error) string {
 	return err.Error()
 }
 
+type deviceProfileMarshalJson struct{}
+
+func (r deviceProfileMarshalJson) httpErrorCode() int {
+	return http.StatusInternalServerError
+}
+
+func (r deviceProfileMarshalJson) isA(err error) bool {
+	panic(METHOD_NOT_ALLOWED)
```

Comment:
Why do we panic? Panic is bad. Why can't we implement this?

Reply:
Since I use StatusInternalServerError when json.Marshal failed, so I refer to other errorconcept structs, they also use the panic error.
https://github.com/edgexfoundry/edgex-go/blob/master/internal/pkg/errorconcept/default.go#L61-L69
```
type internalServerError struct{}

func (r internalServerError) httpErrorCode() int {
	return http.StatusInternalServerError
}

func (r internalServerError) isA(err error) bool {
	panic(METHOD_NOT_ALLOWED)
}
```

And the METHOD_NOT_ALLOWED means:
```
const (
	METHOD_NOT_ALLOWED = "isA should not be invoked, this is to only be used as a default error concept"
)
```
So I just follow the rule, do you have any suggestions?

Reply:
Other has have `isA` implemented:

```
func (r commandNotAssociatedWithDevice) isA(err error) bool {
	_, ok := err.(errors.ErrCommandNotAssociatedWithDevice)
	return ok
}
```
https://github.com/edgexfoundry/edgex-go/blob/master/internal/pkg/errorconcept/command.go#L36

Panic should never be use post start-up, i.e service once service is up an running avoid using panic. In this case it is simple implement `isA` as above rather than `panic`.

Reply:
Got it, added a simple implementation.